test/test_tdmodules.py

```diff
@@ -355,6 +355,10 @@ def test_stateful(self, safe, spec_type, probabilistic, lazy):
             )
             tdmodule = TDSequence(tdmodule1, tdmodule2)
 
+        assert hasattr(tdmodule, "__getitem__")
+        assert tdmodule[0] == tdmodule1
```

Comment:
perhaps `is` instead of `==`? Not sure such behaviour will be robust to any single use case